dev/TEAL.md

```diff
@@ -0,0 +1,219 @@
+---
+numbersections: true
+title: "Algorand Transaction Execution Approval Language"
+date: \today
+abstract: >
+  Algorand allows transactions to be effectively signed by a small program. If the program evaluates to true then the transaction is allowed. This document defines the language and bytcode format.
+---
+
+# Transaction Execution Approval Language (TEAL)
+
+TEAL is a bytecode based stack language that executes inside Algorand transactions to check the parameters of the transaction and approve the transaction as if by a signature.
+
+TEAL programs should be short, at most 1000 bytes including all constants and operations, and run fast as they are run in-line along with signature checking, transaction balance rule checking, and other checks during block assembly and validation.
+
+## The Stack
+
+The stack starts empty and contains values of either uint64 or bytes. (`bytes` implemented in Go as a []byte slice) Most operations act on the stack, popping arguments from it and pushing results to it.
+
+The maximum stack depth is currently 1000.
```

Comment:
Is it 1000, or is it the maximum program size?

Reply:
1000, unrelated constant. It could be a consensus parameter on its own? Or for now I feel like this matters less than size and cost.